axlearn/common/utils.py

```diff
@@ -443,6 +443,11 @@ def with_sharding_constraint(x, shardings):
         return x
     return jax.lax.with_sharding_constraint(x, shardings)
 
+def maybe_shard(x: Tensor, partition_spec: Optional[PartitionSpec]) -> Tensor:
```

Comment:
Is `x` a `Nested[Tensor]`?

Reply:
I think sometimes it is. That is why I did not have a type. I know for `aten_output `it is not a tensor. 

What type do you recommend?